utils/symbol-rawelf.c

```diff
@@ -13,6 +13,96 @@
 #include "utils/utils.h"
 #include "utils/symbol-rawelf.h"
 
+/*
+ *  ELF File Header validation logic.
+ */
+int elf_validation(struct uftrace_elf_data *elf)
+{
+	Elf_Ehdr *ehdr;
+	int eclass, data, version;
+	int res = 0;
+	unsigned long mapped_saddr, mapped_eaddr, mapped_size, addr;
+
+	ehdr = &elf->ehdr;
+
+	// validate ELF Magic.
+	if (memcmp(ehdr, ELFMAG, SELFMAG)) {
+		pr_dbg("ELF Signature not matched\n");
+		res = -1;
+	}
+
+	// validate some field of elf header.
+	eclass = (int) ((unsigned char *) ehdr)[EI_CLASS];
+	data = (int) ((unsigned char *) ehdr)[EI_DATA];
+	version = (int) ((unsigned char *) ehdr)[EI_VERSION];
+
+	if (!(eclass > ELFCLASSNONE && eclass < ELFCLASSNUM)) {
+		pr_dbg("eclass out of range. it must be inside range "\
+				"between %d and %d, but Yours is [%d]\n",
+				ELFCLASSNONE, ELFCLASSNUM, eclass);
+		res = -1;
+	}
+
+	if (!(data > ELFDATANONE && data < ELFDATANUM)) {
+		pr_dbg("data out of range. it must be inside range "\
+				"between %d and %d, but Yours is [%d]\n",
+				ELFDATANONE, ELFDATANUM, data);
+		res = -1;
+	}
+
+	if (!(version > EV_NONE && version < EV_NUM)) {
+		pr_dbg("version out of range. it must be inside range "\
+				"between %d and %d, but Yours is [%d]\n",
+				EV_NONE, EV_NUM, version);
+		res = -1;
+	}
```

Comment:
ditto.

Also it should check the class and endian is same for the host.  IOW raw-ELF code cannot handle 64-bit ELF on 32-bit systems and vice versa.